lib/client/tncon/tncon.go

```diff
@@ -29,95 +28,71 @@ import "C"
 
 import (
 	"fmt"
+	"io"
 	"sync"
 	"unsafe"
 )
 
 var (
-	subscribers      []chan interface{}
-	subscribersMutex sync.Mutex
+	sequenceBuffer *bufferedChannelPipe
+
+	resizeEventSubscribers      []chan struct{}
+	resizeEventSubscribersMutex sync.Mutex
 
 	running           bool = false
 	runningMutex      sync.Mutex
 	runningQuitHandle C.HANDLE
 )
 
-// SequenceEvent is emitted when one or more key sequences are generated. This
-// implementation generally produces many 1-byte events rather than one event
-// per keystroke unless VT sequence translation is enabled.
-type SequenceEvent struct {
-	Sequence []byte
-}
-
-// ResizeEvent is emitted when the window size has been modified. The semantics
-// of this event may vary depending on the current terminal and its flags:
-//  - `cmd.exe` tends not to emit vertical resize events, and horizontal events
-//    have nonsensical height (`Y`) values.
-//  - `powershell.exe` emits events reliably, but height values are still
-//    insane.
-//  - The new Windows Terminal app emits sane events for both horizontal and
-//    vertical resize inputs.
-type ResizeEvent struct {
-	X int16
-
-	// y is the resized height. Note that depending on console mode, this
-	// number may not be sensible and events may not trigger on vertical
-	// resize.
-	Y int16
-}
-
-// writeEvent dispatches an event to all listeners.
-func writeEvent(event interface{}) {
-	subscribersMutex.Lock()
-	defer subscribersMutex.Unlock()
-
-	for _, sub := range subscribers {
-		sub <- event
-	}
+func SequenceReader() io.Reader {
+	return sequenceBuffer
 }
 
 //export writeSequenceEvent
 func writeSequenceEvent(addr *C.char, len C.int) {
```

Comment:
nit: maybe we should drop the `Event` suffix from this now that it's just appending to a buffer